zenodo/modules/profiles/views.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo profiles."""
+
+from __future__ import absolute_import, print_function
+
+import hashlib
+
+from flask import Blueprint, abort, current_app, flash, redirect, \
+    render_template, request
+from flask_babelex import lazy_gettext as _
+from flask_login import current_user, login_required
+from flask_security.confirmable import send_confirmation_instructions
+from invenio_accounts.models import User
+from invenio_db import db
+from invenio_oauthclient.models import UserIdentity
+from invenio_userprofiles.api import current_userprofile
+from six.moves import urllib
+
+from zenodo.modules.profiles.api import ResearcherRecordsSearch
+from zenodo.modules.profiles.forms import ContactOwnerForm
+from zenodo.modules.profiles.models import Profile
+from zenodo.modules.profiles.utils import send_support_email
+
+blueprint = Blueprint(
+    'zenodo_profiles',
+    __name__,
+    template_folder='templates',
+    static_folder='static',
+)
+
+
+@blueprint.route(
```

Comment:
You can move this view handler below the base `/profile/<int:owner_id>` definition, since it's basically a "sub-resource".